com.unity.uiwidgets/Runtime/foundation/debug.cs

```diff
@@ -63,31 +44,59 @@ public static class D {
 
         public static bool debugPrintMouseHoverEvents = false;
 
+        public static bool debugLog = true;
+
         public static HSVColor debugCurrentRepaintColor =
             HSVColor.fromAHSV(0.4f, 60.0f, 1.0f, 1.0f);
 
+        public static int? debugFloatPrecision;
+
+        public static void logError(string message, Exception ex = null) {
+            Debug.LogException(new AssertionError(message: message, innerException: ex));
+        }
+
+        [Conditional("UNITY_ASSERTIONS")]
+        public static void assert(Func<bool> result, Func<string> message = null) {
+            if (debugLog && !result()) {
+                throw new AssertionError(message != null ? message() : "");
+            }
+        }
+
+        [Conditional("UNITY_ASSERTIONS")]
+        public static void assert(bool result, Func<string> message = null) {
+            if (debugLog && !result) {
+                throw new AssertionError(message != null ? message() : "");
+            }
+        }
+
+        public static void setDebugLog(bool _debugLog) {
```

Comment:
we add two APIs:
public static void disableDebugLog() {}
public static void resetDebugLog() {}